packages/yswaps/scripts/libraries/solvers/multicall/CurveSpellEth.ts

```diff
@@ -88,41 +101,55 @@ export class CurveSpellEth implements Solver {
+    if (approveCvx) {
+      console.log('[CurveSpellEth] Approving cvx');
+      const approveCvxTx = await cvx.populateTransaction.approve(zrxCvxAllowanceTarget, constants.MaxUint256);
+      await multicallSwapperSigner.sendTransaction(approveCvxTx);
+      transactions.push(approveCvxTx);
+    }
 
+    console.log('[CurveSpellEth] Executing cvx => weth via zrx');
     const cvxToWethTx: Tx = {
       to: this.zrxContractAddress,
       data: zrxCvxData,
     };
     await multicallSwapperSigner.sendTransaction(cvxToWethTx);
+    transactions.push(cvxToWethTx);
 
-    console.log('[CurveSpellEth] Convert weth to eth');
     const wethBalance = await weth.balanceOf(multicallSwapperAddress);
-    await weth.withdraw(wethBalance);
-
-    console.log('[CurveSpellEth] Convert weth to crvSpellEth');
-    await curveSwap.add_liquidity([wethBalance, 0], 0, true, this.strategyAddress, { value: wethBalance });
+    console.log('[CurveSpellEth] Total WETH balance is', utils.formatEther(wethBalance));
+
+    const approveWeth = (await weth.allowance(multicallSwapperAddress, curveSwap.address)).lt(wethBalance);
+    if (approveWeth) {
+      console.log('[CurveSpellEth] Approving weth');
+      const approveWethTx = await weth.populateTransaction.approve(curveSwap.address, constants.MaxUint256);
+      await multicallSwapperSigner.sendTransaction(approveWethTx);
+      transactions.push(approveWethTx);
+    }
+
+    console.log('[CurveSpellEth] Converting eth to crvSpellEth');
+    const curveCalculatedTokenAmountOut = await curveSwap.calc_token_amount([wethBalance, 0]);
+    const curveCalculatedTokenMinAmountOut = curveCalculatedTokenAmountOut.sub(curveCalculatedTokenAmountOut.mul(3).div(100)); // 3% slippage
+    const addLiquidityTx = await curveSwap.populateTransaction.add_liquidity(
+      [wethBalance, 0],
+      curveCalculatedTokenMinAmountOut,
+      false,
+      this.strategyAddress
+    );
+    await multicallSwapperSigner.sendTransaction(addLiquidityTx);
+    transactions.push(addLiquidityTx);
 
-    // Create txs for multichain swapper
-    const transactions: PopulatedTransaction[] = [];
+    const amountOut = await crvSpellEth.balanceOf(this.strategyAddress);
+    console.log('[CurveSpellEth] Final crvSpellEth balance is', utils.formatEther(amountOut));
 
-    // zrx trades
-    if (approveCrv) transactions.push(await crv.populateTransaction.approve(zrxCrvAllowanceTarget, constants.MaxUint256));
-    transactions.push(crvToWethTx);
-    if (approveCvx) transactions.push(await cvx.populateTransaction.approve(zrxCvxAllowanceTarget, constants.MaxUint256));
-    transactions.push(cvxToWethTx);
+    if (amountOut.eq(0)) throw new Error('No crvSpellEth tokens were received');
 
-    // Withdraw eth
-    transactions.push(await weth.populateTransaction.withdraw(wethBalance));
+    const minAmountOut = amountOut.sub(amountOut.mul(3).div(100)); // 3% slippage
```

Comment:
should we use `curveCalculatedTokenMinAmountOut` instead of recalculating the minAmountOut with result of weth => crvSpellEth? If we got 2% of slippage on the swap to crvSpellEth and we recalculate minAmountOut again with a 3% slippage we would be making the slippage bigger?